jsonchecker/__init__.py

```diff
@@ -35,15 +35,15 @@ def checker(self, seq, fname):
 
     def check_directory(self, directory):
         """Check one directory."""
-        if directory == '.':
-            directory = os.getcwd()
+        directory = os.path.abspath(directory)
         if os.path.isdir(directory):
+            last_directory = os.path.basename(os.path.normpath(directory))
+            if last_directory.startswith("."):
+                return
             files = [os.path.join(directory, fname) for fname in os.listdir(directory)]
         else:
             files = [directory]
         for fname in files:
-            if fname.startswith('.'):
```

Comment:
the current version skips dot-directories like .git / .hg, but it doesnt skip dot-files.  Skipping dot files is less critical, as they are usually smaller.